repo2docker/buildpacks/repo2docker-entrypoint

```diff
@@ -3,6 +3,9 @@
 # we use a login shell to get a fully configured shell environment
 # (e.g. sourcing /etc/profile.d, ~/.bashrc, and friends)
 if [[ ! -z "${R2D_ENTRYPOINT:-}" ]]; then
+    if [[ ! -x "$R2D_ENTRYPOINT" ]]; then
+        chmod u+x "$R2D_ENTRYPOINT"
```

Comment:
I'm undecided on whether `u+x` or `a+x` is better

Reply:
I specified only u+x because ultimately that is the minimum action to fix this use case.

It might also be helpful to extend to a+x but (1) I'm not sure when/why that might be useful, and (2) think it is more likely to have unforeseen side effects compared to just leaving as u+x.

I was thinking that unless anyone knows an obscure operating system or a case where a+x will succeed but u+x somehow doesn't fix the original problem, it could be left as u+x until someone discovers that it needs to be extended!

BUT for comparison, start_script is given a+x in the docker build. That is also carried out as NB_USER, although that is within the docker build which may be different.

As another comparison, in docker-stacks, the [fix-permissions](https://github.com/jupyter/docker-stacks/blob/master/base-notebook/fix-permissions) script seems to prefer permissions centred on the NB_GID group (g+rwX).

Any of these approaches works from my point of view. I would suggest keep it minimal with u+x but welcome your decision :)

Reply:
:+1: I'll let someone else check and merge